packages/babel-core/src/config/files/configuration.ts

```diff
@@ -188,7 +188,7 @@ const readIgnoreConfig = makeStaticFileCache((filepath, content) => {
   const ignoreDir = path.dirname(filepath);
   const ignorePatterns = content
     .split("\n")
-    .map<string>(line => line.replace(/#(.*?)$/, "").trim())
+    .map<string>(line => line.replace(/#.*$/, "").trim())
```

Comment:
Aside: this should have been `/^#.*$/`, now we are removing everything after `#`. In Babel 8 we can probably simplify the manipulations here with one global regex replace and split.

Reply:
Isn't it intentional? e.g.
```gitignore
./foo.js # We want to ignore foo.js
```

Reply:
The `.gitignore` only interprets `#` starting from the beginning as the start of a comment ([ref](https://git-scm.com/docs/gitignore#_pattern_format)), because `#` can form a valid filename, at least on macOS. Currently it is impossible to ignore a `#`-containing file on `.babelignore` due to the regex here, though we can always specify it on Babel options. I think the motivation of `.babelignore` is to start from a copy of `.gitignore` and amend more files that one does not intend to compile, so it would be great if we can follow the `.gitignore` syntax.